crates/sui-indexer/src/framework/fetcher.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use sui_rest_api::{CheckpointData, Client};
+use sui_types::messages_checkpoint::CheckpointSequenceNumber;
+use tracing::{info, warn};
+
+pub struct CheckpointFetcher {
+    client: Client,
+    last_downloaded_checkpoint: Option<CheckpointSequenceNumber>,
+    highest_known_checkpoint: CheckpointSequenceNumber,
+    sender: mysten_metrics::metered_channel::Sender<CheckpointData>,
+    //TODO add a checkpoint cache
```

Comment:
for my education can you spend few words on what this `TODO` means?

Reply:
i don't recall, i'll remove the comment